File: Apriori/Apriori_R/Apriori (PK).R
# Apriori

dataset <- read.csv('Market_Basket_Optimisation.csv', header = FALSE)

# Data Preprocessing
# Transfer Dataset into Sparse Martix
#install.packages('arules')
library(arules)
dataset <- read.transactions('Market_Basket_Optimisation.csv', sep = ',', rm.duplicates = TRUE) #Removing Duplicates 
summary(dataset)
itemFrequencyPlot(dataset, topN = 100)

# Training Apriori on the dataset
# Support = item sold per day*days of week / Total transaction in a week = 3*7/7500
rules <- apriori(dataset, parameter = list(support = 0.003, confidence = 0.4)) 

# Visualize the Apriori Model
inspect(sort(rules, by = 'lift')[1:10])

File: Apriori/Apriori_Python/Apriori (PK).py
# Apriori
# Import Libraries
import numpy as np
import matplotlib.pyplot as plt
import pandas as pd

# Import the Dataset
dataset = pd.read_csv('Market_Basket_Optimisation.csv', header = None)

transactions = []
for i in range(0, 7501):
    transactions.append([str(dataset.values[i,j]) for j in range(0, 20)])
    
# Training our Apriori Model
from apyori import apriori
rules = apriori(transactions, min_support = 0.003, min_confidence = 0.2, min_lift = 3, min_length = 2)

# Visualize the results
results = list(rules)